infra/main.bicep

```diff
@@ -345,6 +358,114 @@ module backend 'core/host/appservice.bicep' = {
   }
 }
 
+// Azure container apps resources
+
+// identity for pulling images from ACR
+module acaIdentity 'core/security/aca-identity.bicep' = if (deploymentTarget == 'containerapps') {
+  name: 'aca-identity'
+  scope: resourceGroup
+  params: {
+    identityName: acaIdentityName
+    location: location
+  }
+}
+
+module containerApps 'core/host/container-apps.bicep' = if (deploymentTarget == 'containerapps') {
+  name: 'container-apps'
+  scope: resourceGroup
+  params: {
+    name: 'app'
+    location: location
+    workloadProfile: azureContainerAppsWorkloadProfile
+    containerAppsEnvironmentName: acaManagedEnvironmentName
+    containerRegistryName: '${containerRegistryName}${resourceToken}'
+    logAnalyticsWorkspaceResourceId: monitoring.outputs.logAnalyticsWorkspaceId
+  }
+}
+
+// The application frontend
+module acaBackend 'core/host/container-app-upsert.bicep' = if (deploymentTarget == 'containerapps') {
+  name: 'aca-web'
+  scope: resourceGroup
+  dependsOn: [
+    containerApps
+    acaIdentity
+  ]
+  params: {
+    name: !empty(backendServiceName) ? backendServiceName : '${abbrs.webSitesContainerApps}backend-${resourceToken}'
+    location: location
+    identityName: (deploymentTarget == 'containerapps') ? acaIdentityName : ''
+    exists: webAppExists
+    workloadProfile: azureContainerAppsWorkloadProfile
+    containerRegistryName: (deploymentTarget == 'containerapps') ? containerApps.outputs.registryName : ''
+    containerAppsEnvironmentName: (deploymentTarget == 'containerapps') ? containerApps.outputs.environmentName : ''
+    identityType: 'UserAssigned'
+    tags: union(tags, { 'azd-service-name': 'backend' })
+    targetPort: 8000
+    containerCpuCoreCount: '1.0'
+    containerMemory: '2Gi'
+    allowedOrigins: [ allowedOrigin ]
+    env: {
+      AZURE_STORAGE_ACCOUNT: storage.outputs.name
```

Comment:
This block of code here repeats the env variables used earlier, which makes me worry that one will get updated and not the other. It also makes the file much longer.

Could you extract the env variables into an object, like appEnvVariables, that gets passed into each of them?

I'm also debating whether there should be a separate Bicep module for the app service stuff vs the container app stuff but don't know if that'll make things cleaner or not. Would like @tonybaloney's thoughts on what would be the most compatible with eventual move to Azure Verified Modules.

Reply:
appEnvVariables is added and shared between appservice and aca resource